modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonNextgenClientCodegen.java

```diff
@@ -443,12 +449,20 @@ private String getPydanticType(CodegenParameter cp,
                     fieldCustomization.add("multiple_of=" + cp.getMultipleOf());
                 }
 
+                if (floatStrictType) {
```

Comment:
This code path is used for json schema number (float + int) and float only processing. Number should allow both in.

How about:
- not adding the strict info when it is number
- adding it when it is float only
- adding tests of both of those cases with and without strict on
- adding a test that verifies the 1.0 ingestion fix

Reply:
It's an option so users can choose whether they want float in strict type or not.

Reply:
Right but that option should only apply to floats right? The way it is now looks like you are applying flat strict when the spec has 'type: number' which should allow in float and int. Did I misunderstand something?

Reply:
```diff
-    number: confloat(strict=True, le=543.2, ge=32.1) = ...
-   float: Optional[confloat(strict=True, le=987.6, ge=54.3)] = None
-    double: Optional[confloat(strict=True, le=123.4, ge=67.8)] = None
+    number: confloat(le=543.2, ge=32.1) = ...
+    float: Optional[confloat(le=987.6, ge=54.3)] = None
+    double: Optional[confloat(le=123.4, ge=67.8)] = None
```

As illustrated in the sample diff as part of this change, the option (when set to false) simply removes `strict=True`.

Reply:
Why are you applying strict float to type number with no format? Won't it throw a validation error if an integer input is given to it? Integer input is valid for that definition.

Reply:
> Why are you applying strict float to type number with no format?

It's a user decision - they can switch on or off strict mode via an option.

You can find more about why some users prefer strict mode in https://docs.pydantic.dev/blog/pydantic-v2/#strict-mode.